test/verifyexclude/VerifyExcludeTest.gmk

```diff
@@ -0,0 +1,190 @@
+	echo $@ passed at `date` > $@
+
+$(BUILDTESTDIR)/verifyexclude.good.ok: \
+                $(JTREG_IMAGEDIR)/lib/jtreg.jar \
+		$(JTREG_IMAGEDIR)/lib/javatest.jar
+	$(RM) $(@:%.ok=%/work) $(@:%.ok=%/report)
+	$(MKDIR) -p $(@:%.ok=%)
+	$(JDKHOME)/bin/java \
+		-Djavatest.regtest.showCmd=true \
+		-jar $(JTREG_IMAGEJARDIR)/jtreg.jar \
+		-w:$(@:%.ok=%/work) \
+		-r:$(@:%.ok=%/report) \
+		-exclude:$(TESTDIR)/verifyexclude/ProblemList/good.txt \
+		--verify-exclude \
+		$(TESTDIR)/verifyexclude \
+		> $(@:%.ok=%/log) \
+	    || (cat $(@:%.ok=%/log) ; exit 1)
+	echo $@ passed at `date` > $@
+
+$(BUILDTESTDIR)/verifyexclude.duplicate.ok: \
+                $(JTREG_IMAGEDIR)/lib/jtreg.jar \
+		$(JTREG_IMAGEDIR)/lib/javatest.jar
+	$(RM) $(@:%.ok=%/work) $(@:%.ok=%/report)
+	$(MKDIR) -p $(@:%.ok=%)
+	$(JDKHOME)/bin/java \
+		-Djavatest.regtest.showCmd=true \
+		-jar $(JTREG_IMAGEJARDIR)/jtreg.jar \
+		-w:$(@:%.ok=%/work) \
+		-r:$(@:%.ok=%/report) \
+		-exclude:$(TESTDIR)/verifyexclude/ProblemList/duplicate.txt \
+		--verify-exclude \
+		$(TESTDIR)/verifyexclude \
+		> $(@:%.ok=%/log) \
+	    && (cat $(@:%.ok=%/log) ; exit 1) \
+		|| (exit 0)
+	$(GREP) -s "Exclude file cannot contain duplicate entries." $(@:%.ok=%/log)
+	echo $@ passed at `date` > $@
+
+$(BUILDTESTDIR)/verifyexclude.match.format.1.ok: \
+                $(JTREG_IMAGEDIR)/lib/jtreg.jar \
+		$(JTREG_IMAGEDIR)/lib/javatest.jar
+	$(RM) $(@:%.ok=%/work) $(@:%.ok=%/report)
+	$(MKDIR) -p $(@:%.ok=%)
+	$(JDKHOME)/bin/java \
+		-Djavatest.regtest.showCmd=true \
+		-jar $(JTREG_IMAGEJARDIR)/jtreg.jar \
+		-w:$(@:%.ok=%/work) \
+		-r:$(@:%.ok=%/report) \
+		-match:$(TESTDIR)/verifyexclude/ProblemList/format.1.txt \
+		--verify-exclude \
+		$(TESTDIR)/verifyexclude \
+		> $(@:%.ok=%/log) \
+	    && (cat $(@:%.ok=%/log) ; exit 1) \
+		|| (exit 0)
+	$(GREP) -s "Must follow:" $(@:%.ok=%/log)
+	$(GREP) -s "The fully qualified test must exists." $(@:%.ok=%/log) && exit 1 || exit 0
+	echo $@ passed at `date` > $@
+
+$(BUILDTESTDIR)/verifyexclude.shortform.ok: \
+                $(JTREG_IMAGEDIR)/lib/jtreg.jar \
+		$(JTREG_IMAGEDIR)/lib/javatest.jar
+	$(RM) $(@:%.ok=%/work) $(@:%.ok=%/report)
+	$(MKDIR) -p $(@:%.ok=%)
+	$(JDKHOME)/bin/java \
+		-Djavatest.regtest.showCmd=true \
+		-jar $(JTREG_IMAGEJARDIR)/jtreg.jar \
+		-w:$(@:%.ok=%/work) \
+		-r:$(@:%.ok=%/report) \
+		-exclude:$(TESTDIR)/verifyexclude/ProblemList/format.1.txt \
+		-i \
+		$(TESTDIR)/verifyexclude \
+		> $(@:%.ok=%/log) \
+	    && (cat $(@:%.ok=%/log) ; exit 1) \
+		|| (exit 0)
+	$(GREP) -s "Must follow:" $(@:%.ok=%/log)
+	echo $@ passed at `date` > $@
+
+TESTS.jtreg += \
+	$(BUILDTESTDIR)/verifyexclude.duplicate.ok \
+	$(BUILDTESTDIR)/verifyexclude.format.1.ok \
+	$(BUILDTESTDIR)/verifyexclude.format.2.ok \
+	$(BUILDTESTDIR)/verifyexclude.good.ok \
+	$(BUILDTESTDIR)/verifyexclude.id.ok \
+	$(BUILDTESTDIR)/verifyexclude.exist.ok \
+    $(BUILDTESTDIR)/verifyexclude.match.format.1.ok \
+    $(BUILDTESTDIR)/verifyexclude.shortform.ok
```

Comment:
Something may have gone wrong with the indentation here

Reply:
fixed